pkg/gateway/gateway.go

```diff
@@ -566,16 +565,13 @@ func (g *gatewayModule) setupRouting(wlID string, fqdn string, backends []string
 	servers := make([]Server, len(backends))
 	for idx, backend := range backends {
 		if TLSPassthrough {
-			if err := zos.Backend(backend).Valid(true); err != nil {
-				return errors.Wrap(err, "couldn't validate backend host")
-			}
-			u, err := url.Parse(backend)
-			log.Debug().Str("hostname", u.Host).Str("backend", backend).Msg("tls passthrough")
+			host, err := zos.Backend(backend).Parse(true)
```

Comment:
this is still wrong because instead it can be like 
```go 
for idx, backend := range backends {
   u, err := zos.Backend(backend).Parse(TLSPassthrout)
    // check error
   if TLSPassthrough {
   servers[ids] = Server{ Address: u} 
   } else {
   servers[idx] = Server{ Url: u} 
  } 
}

```